woodwork/utils.py

```diff
@@ -298,3 +299,149 @@ def _parse_logical_type(logical_type, name):
+                    elif obj_idx != table_idx:
+                        raise IndexError(f'Cannot concat dataframes with different Woodwork {idx_type} columns.')
+                    elif axis in {1, 'columns'} and obj_idx == table_idx:
+                        # Do not include the column from the dataframe to avoid duplicates columns
+                        drop_cols.append(table_idx)
+
+            # To avoid changing input objects, create a new DataFrame without the index columns
+            if drop_cols:
+                obj_to_add = obj.ww.drop(columns=drop_cols)
+            ww_columns = obj_to_add.ww.schema.columns
+        elif isinstance(obj.ww.schema, ww.column_schema.ColumnSchema):
+            ww_columns = {obj.name: obj.ww.schema}
+
+        concat_objs.append(obj_to_add)
+
+        # Compile the typing information per column
+        for name, col_schema in ww_columns.items():
+            # Do not look at typing information for duplicate columns.
+            # Means that index columns will only have first occurrance's typing information
+            if name in col_names_seen:
+                continue
+            logical_types[name] = col_schema.logical_type
+            semantic_tags[name] = col_schema.semantic_tags - {'time_index'} - {'index'}
+            col_metadata[name] = col_schema.metadata
+            col_descriptions[name] = col_schema.description
+            use_standard_tags[name] = col_schema.use_standard_tags
+
+            col_names_seen.add(name)
+
+    # Perform concatenation with the correct library
+    dd = import_or_none('dask.dataframe')
+    ks = import_or_none('databricks.koalas')
+
+    lib = pd
+    if ks and isinstance(obj, (ks.Series, ks.DataFrame)):
+        lib = ks
+    elif dd and isinstance(obj, (dd.Series, dd.DataFrame)):
+        lib = dd
+
+    combined_df = lib.concat(concat_objs, axis=axis, join=join)
+
+    # Cannot initialize Woodwork on series output
+    # --> we can handle if we really want to we'll just need to account for this case when collecting typging info
+    if not isinstance(combined_df, lib.DataFrame):
+        warnings.warn('Woodwork not initialized on the output of ww.concat', ww.exceptions.WoodworkNotInitWarning)
+        return combined_df
+
+    # Inner joins can lose columns, invalidating the typing information we collected ahead of time
+    if join == 'inner':
+        combined_cols = set(combined_df.columns)
+        if time_index not in combined_cols:
+            time_index = None
+        if index not in combined_cols:
+            index = None
+
+        def filter_cols(typing_dict):
+            return {name: typing_val for name, typing_val in typing_dict.items() if name in combined_cols}
+
+        logical_types = filter_cols(logical_types)
+        semantic_tags = filter_cols(semantic_tags)
+        col_metadata = filter_cols(col_metadata)
+        col_descriptions = filter_cols(col_descriptions)
+        use_standard_tags = filter_cols(use_standard_tags)
```

Comment:
I'm considering pulling all of this logic into a single for-loop in order to not have 5 separate `filter_cols` calls that each can loop over the number of columns in the resulting dataframe.

In general, this `concat` function feels very messy to me. All of the disparate typing variables make it hard to pull logic into separate functions without having to return a length-7 tuple, and it's hard to separate the portions of this code that are changing data vs compiling the typing information.

Reply:
@thehomebrewnerd made the simplification changes (fixing `axis=1`, `join=outer`, and only allowing one index and time index colum to be specified across the inputs) along with some other suggestions that have simplified this util function. 

Let me know what you think!